### ISSUE TITLE
Angular parser breaks ngClass construct

### ISSUE THREAD

**Issue by spitfjre** [prettier/prettier#5367]
Tried out the new angular parser. After formatting our very large code base. I stumbled over this, because formatting this with prettier produced different results in different iterations.

**Prettier 1.15.1**

**Input:**
```angular
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    'btn-success': (dialog$ | async).level === dialogLevelEnum.SUCCESS,
    'btn-warning': (dialog$ | async).level === dialogLevelEnum.WARNING,
    'btn-svg': (dialog$ | async).level === dialogLevelEnum.DANGER
  }">
  {{(dialog$ | async).btnTxt}}
</button>
```

**Output (1st iteration):**
```angular
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    '-success\': ': (dialog$ | async).level === dialogLevelEnum.SUCCESS,
    '-warning\': ': (dialog$ | async).level === dialogLevelEnum.WARNING,
    '-svg\': ': (dialog$ | async).level === dialogLevelEnum.DANGER
  }"
>
  {{ (dialog$ | async).btnTxt }}
</button>
```

**Output (2nd iteration):**
```angular
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    'cess\': \': ': (dialog$ | async).level === dialogLevelEnum.SUCCESS,
    'ning\': \': ': (dialog$ | async).level === dialogLevelEnum.WARNING,
    '\': \': ': (dialog$ | async).level === dialogLevelEnum.DANGER
  }"
>
  {{ (dialog$ | async).btnTxt }}
</button>
```

**Output (3rd iteration):**
```angular
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    '\': \': \': ': (dialog$ | async).level === dialogLevelEnum.SUCCESS,
    '\': \': \': ': (dialog$ | async).level === dialogLevelEnum.WARNING,
    '\': \': ': (dialog$ | async).level === dialogLevelEnum.DANGER
  }"
>
  {{ (dialog$ | async).btnTxt }}
</button>
```

**Output (4th iteration):**
```angular
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    ': \': \': ': (dialog$ | async).level === dialogLevelEnum.SUCCESS,
    ': \': \': ': (dialog$ | async).level === dialogLevelEnum.WARNING,
    '\': \': ': (dialog$ | async).level === dialogLevelEnum.DANGER
  }"
>
  {{ (dialog$ | async).btnTxt }}
</button>
```

**Output (5th iteration):**
```angular
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    ' \': \': ': (dialog$ | async).level === dialogLevelEnum.SUCCESS,
    ' \': \': ': (dialog$ | async).level === dialogLevelEnum.WARNING,
    '\': \': ': (dialog$ | async).level === dialogLevelEnum.DANGER
  }"
>
  {{ (dialog$ | async).btnTxt }}
</button>
```

**Output (6th iteration - stable from now on):**
```angular
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    '\': \': ': (dialog$ | async).level === dialogLevelEnum.SUCCESS,
    '\': \': ': (dialog$ | async).level === dialogLevelEnum.WARNING,
    '\': \': ': (dialog$ | async).level === dialogLevelEnum.DANGER
  }"
>
  {{ (dialog$ | async).btnTxt }}
</button>
```

**Expected behavior:**
Not to break code.

**Comment by ikatyang** [comment]
Yeah, I guess I did something wrong in [angular-estree-parser](https://github.com/ikatyang/angular-estree-parser).

**Comment by michaeljota** [comment]
@ikatyang This seems to be related to the parenthesis around the async property.

[Link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeARgVxjaAdKAAkJgE8AHOAXjxCx3xAOIAowAbASzAGsBKGuHYQAznAAinAIbCA5iz61mhDlJEjB6GES1QAtOQBOnALZTDpJUUIBtKLIDC7NSIC6g4AHJdekZjBgcOqeSITscABucOyEVHGEACbScgAykdEAolCYJgB0AMoAqg4OGfn5ADTe2noA7uZQnPYhhCxJMhCyACSEAD6EaqRQYHy54VExcVSJyZ1pE1k5uQDqAIIASgByAJKbAOJVPiIRsi1ts919AyJDI2Ppk-HtqQ+LeeKr+xnrygC+tAA+ZTAYDnDqXfqDYajXQAFQAHjBfv8oKgAPT0XBQAEgCogCDkGCcaAiZCgcyGCC1AAK5gQpJQUgiEE4CVxdEMUl4cBg+XIXKasmQMEMmDgeIAFjATOxlhLOPARPzAvl6QrOBEFaRkOB1OymmJDDBqZzZGZkAAzGRiPEAKxE8IAQpzubypCY4CkmnBLdbxSB7fD8oLwgBFTAQeC+9g2kD8wyGnVSeyYZyGdlGJowZasmAS5AADgADHijKI4MtOeQdUYgnBDFF2YY4ABHTCcZsmqRmqTR2NiEycYWi-0iENwcORn1IK0x-0wKToHMJPPIABMeJFUk4XHsDggJnNKCg0B9eMwYlhi4Zs7EyKAA)

* Input:
```html
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    'btn-success': level === dialogLevelEnum.SUCCESS,
    'btn-warning': isLevelSuccess(dialog$ | async),
    'btn-danger': (dialog$ | async).level === dialogLevelEnum.DANGER
  }">
  {{(dialog$ | async).btnTxt}}
</button>
```

* Output: 
```html
<button
  type="button"
  (click)="closeDialog()"
  class="btn btn-primary"
  [ngClass]="{
    'btn-success': level === dialogLevelEnum.SUCCESS,
    'btn-warning': isLevelSuccess(dialog$ | async),
    '-danger\': ': (dialog$ | async).level === dialogLevelEnum.DANGER
  }"
>
  {{ (dialog$ | async).btnTxt }}
</button>
```

**Comment by ikatyang** [comment]
The fix is available in [`1.15.2`](https://github.com/prettier/prettier/blob/master/CHANGELOG.md#1152).
